File: contest/aizu/itp1/itp1-1-c/ITP1_1_C.cpp
#include <cstdio>

namespace {
    class Rectangle {
    public:
        Rectangle(int width, int height) {
            this->width = width;
            this->height = height;
        }
        int getArea() {
            return width * height;
        }
        int getPerimeter() {
            return 2 * (width + height);
        }
    private:
        int height;
        int width;
    };
}

int main(int argc, char** argv) {
    int width;
    int height;
    std::scanf("%d %d", &width, &height);
    Rectangle rectangle(width, height);
    std::printf("%d %d\n",
                rectangle.getArea(),
                rectangle.getPerimeter());
    return 0;
}

File: contest/aizu/itp1/itp1-1-c/ITP1_1_C.hs
#!/usr/bin/env runhaskell
module Main (main) where

main :: IO ()
main = do
  interact getString
  where
    getString :: String -> String
    getString x = (unwords . getWords . words) x ++ "\n"
      where
      getWords :: [String] -> [String]
      getWords x = (map show . getInts . map read) x
        where
        getInts :: [Int] -> [Int]
        getInts (width : height : []) = area : perimeter : []
          where
          rectangle = Rectangle width height
          area = getArea rectangle
          perimeter = getPerimeter rectangle

class HasArea a where
  getArea :: a -> Int
class HasPerimeter a where
  getPerimeter :: a -> Int
data Rectangle = Rectangle Int Int
instance HasArea Rectangle where
  getArea (Rectangle width height) = width * height
instance HasPerimeter Rectangle where
  getPerimeter (Rectangle width height) = 2 * (width + height)
